OpenQA/Isotovideo/Runner.pm

```diff
@@ -0,0 +1,80 @@
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Isotovideo::Runner;
+use Mojo::Base -base, -signatures;
+
+use OpenQA::Isotovideo::Utils qw(checkout_git_repo_and_branch checkout_git_refspec);
+
+use bmwqemu;
+use autotest;
+
+use Mojo::File 'path';
+use Try::Tiny;
+
+use constant AUTOINST_STATUSFILE => 'autoinst-status.json';    # XXX
+
+
+# a run is comprised of these individual resources
+has [qw(casedir needles_dir productdir test_git_hash)] => undef;
+
+sub new ($class, %args) {
+    my $self = $class->SUPER::new(
+        casedir => path($args{casedir}),
+        needles_dir => $args{needles_dir},
+        productdir => path($args{productdir}),
+        test_git_hash => undef,
+    );
```

Comment:
```suggestion
sub new ($class, @args) {
    my $self = $class->SUPER::new(@args);
```

Reply:
That would mean the caller has to provide `path` objects, though.